college-management/Contextos/ContextoCargos.cs

```diff
@@ -14,12 +16,422 @@ public ContextoCargos(BaseDeDados baseDeDados,
+            cargo = BaseDeDados.Cargos.ObterPorNome(nomeDoCargo);
+        }
+
+        else if (menuPesquisa.OpcaoEscolhida is 2)
+        {
+            var id = inputPesquisa.ObterEntrada("Id");
+            cargo = BaseDeDados.Cargos.ObterPorId(id);
+        }
+
+        if (cargo == null)
+        {
+            inputPesquisa.LerEntrada("Cargo",
+                                     "Cargo não encontrado.");
+            return;
+        }
+
+        ExibirDetalhesCargo(cargo, detalhesView);
+        Console.ReadKey();
+    }
+
+
+    List<string> SelecaoDePermissao()
+    {
+        List<string> permissoes = new List<string>();
+
+        var propriedades = typeof(PermissoesAcesso).GetFields();
+        string[] nomePropriedades = new string[propriedades.Length];
+        int index = 0;
+
+        for (int i = 0; i < propriedades.Length; i++)
+        {
+            nomePropriedades[i] = propriedades[i].Name;
+            index++;
+        }
+        index = 0;
+        
+
+        while(index < nomePropriedades.Length)
+        {
+            MenuView menuView = new 
+                MenuView("Permissões", 
+                "\t\tPermissões\n\n\n", 
+                nomePropriedades);
+
+            menuView.ConstruirLayout();
+            menuView.Exibir();
+            menuView.LerEntrada();
+
+            int opcao = menuView.OpcaoEscolhida-1;
+
+            if (opcao >= nomePropriedades.Length || opcao < 0) break;
+
+            if (!permissoes.Contains
+                (nomePropriedades
+                [opcao]))
+
+                permissoes.Add(nomePropriedades[opcao]);
+
+
+
+            if (ConfirmarEscolha($"Adicionou {permissoes.Last()}" +
+                $", deseja adicionar mais permissões?\n"))
+
+                index = opcao;
+
+            else break;
+
+
+        }
+
+        Console.Clear();
+        return permissoes;
+        
+    }
+
+
+    string TelaExclusao(InputView inputView)
+    {
+
+        inputView.LerEntrada("name", "Insira o nome do cargo a ser excluido do banco de dados: ");
+
+        var nomeCargo = inputView.ObterEntrada("name");
+
+        return nomeCargo;
+    }
+
+    string SelecionaCargoParaEdicao(InputView inputView)
+    {
+
+        inputView.LerEntrada("name", "Insira o nome do cargo a ser editado: ");
+
+        var cargoId = inputView.ObterEntrada("name");
+
+        return cargoId;
+    }
+
+
+    Cargo TelaDeEdicao(Cargo cargoAtual, InputView inputView)
+    {
+
+        inputView.LerEntrada("nome",
+            $"Nome Atual: {cargoAtual.Nome}\n\nEscolha um novo nome: ");
+
+        if (ValidaEntrada(inputView, "nome"))
+        {
+
+            var permissoes = SelecaoDePermissao();
+
+            if (permissoes.Any())
+                cargoAtual.Permissoes = permissoes;
+
+            else return null;
+
+            cargoAtual.Nome = inputView.ObterEntrada("nome");
+
+            return cargoAtual;
+        }
+
+        return null;
+    }
```

Comment:
A forma na qual esse método foi implementado faz edições diretamente no cargo, e como ``Cargo`` é uma classe ([*reference type*](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types)), ``cargoAtual`` é uma direta referência ao cargo alvo a ser editado. Isso significa que edições são feitas antes mesmo do usuário autorizá-las.
```suggestion
    // Marcar como nullable é sempre boa prática para melhor leitura de código e para evitar bugs.
    Cargo? TelaDeEdicao(Cargo cargoAtual, InputView inputView)
    {
        inputView.LerEntrada("nome",
            $"Nome Atual: {cargoAtual.Nome}\n\nEscolha um novo nome: ");

        if (ValidaEntrada(inputView, "nome"))
        {
            List<string> permissoes = SelecaoDePermissao();
            if (!permissoes.Any()) return null;
            
            return new Cargo(inputView.ObterEntrada("nome"), permissoes); // Retorna um novo cargo invés de usar o existente.
        }

    return null;
}
```

Reply:
> Única coisa estranha que percebi foi que o cargo Alunos não tem o usuário teste, mesmo ele tendo o cargo Alunos.

Acredito que isso seja problema na hora de seedar o "banco"


> A forma na qual esse método foi implementado faz edições diretamente no cargo, e como Cargo é uma classe ([reference type](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/reference-types)), cargoAtual é uma direta referência ao cargo alvo a ser editado. Isso significa que edições são feitas antes mesmo do usuário autorizá-las.

Sim, eu acabei esquecendo desse erro que identifiquei faz bastante tempo. Valeu pela resposta implemente ela no commit [1060205](https://github.com/Os-Derivados/college-management/pull/46/commits/106020521eb28a0e2ef600a149c6342db79820fa) com uma pequena alteração.